src/app/modules/submission/email-submission/email-submission.controller.ts

```diff
@@ -227,7 +227,10 @@ export const handleEmailSubmission: RequestHandler<
     parsedResponses,
     submission,
     attachments,
-    autoReplyData: emailData.autoReplyData,
+    autoReplyData:
+      authType === AuthType.CP
+        ? emailData.maskedAutoReplyData
+        : emailData.autoReplyData,
```

Comment:
personally I would rather have just one `autoReplyData` property, which would fulfil the intention of encapsulation because everything outside doesn't have to care about the internals of which `autoReplyData` is correct. this may mean you have to initialise the `emailData` object with the authType of the form, so when you get the autoreply data, it knows whether to return the masked or unmasked version. @liangyuanruo any thoughts?

Reply:
yes, i think that would be helpful to encapsulate business logic within the class